mmedit/models/restorers/real_basicvsr.py

```diff
@@ -0,0 +1,190 @@
+        pixel_loss (dict, optional): Config for the pixel loss. Default: None.
+        cleaning_loss (dict, optional): Config for the image cleaning loss.
+            Default: None.
+        perceptual_loss (dict, optional): Config for the perceptual loss.
+            Default: None.
+        is_use_sharpened_gt_in_pixel (bool, optional): Whether to use the image
+            sharpened by unsharp masking as the GT for pixel loss.
+            Default: False.
+        is_use_sharpened_gt_in_percep (bool, optional): Whether to use the
+            image sharpened by unsharp masking as the GT for perceptual loss.
+            Default: False.
+        is_use_sharpened_gt_in_gan (bool, optional): Whether to use the
+            image sharpened by unsharp masking as the GT for adversarial loss.
+            Default: False.
+        is_use_ema (bool, optional): When to apply exponential moving average
+            on the network weights. Default: True.
+        train_cfg (dict): Config for training. Default: None.
+            You may change the training of gan by setting:
+            `disc_steps`: how many discriminator updates after one generate
+            update;
+            `disc_init_steps`: how many discriminator updates at the start of
+            the training.
+            These two keys are useful when training with WGAN.
+        test_cfg (dict): Config for testing. Default: None.
+        pretrained (str): Path for pretrained model. Default: None.
+    """
+
+    def __init__(self,
+                 generator,
+                 discriminator=None,
+                 gan_loss=None,
+                 pixel_loss=None,
+                 cleaning_loss=None,
+                 perceptual_loss=None,
+                 is_use_sharpened_gt_in_pixel=False,
+                 is_use_sharpened_gt_in_percep=False,
+                 is_use_sharpened_gt_in_gan=False,
+                 is_use_ema=True,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+
+        super().__init__(generator, discriminator, gan_loss, pixel_loss,
+                         perceptual_loss, is_use_sharpened_gt_in_pixel,
+                         is_use_sharpened_gt_in_percep,
+                         is_use_sharpened_gt_in_gan, is_use_ema, train_cfg,
+                         test_cfg, pretrained)
+
+        self.cleaning_loss = build_loss(
+            cleaning_loss) if cleaning_loss else None
+
+    def train_step(self, data_batch, optimizer):
+        """Train step.
+
+        Args:
+            data_batch (dict): A batch of data.
+            optimizer (obj): Optimizer.
+
+        Returns:
+            dict: Returned output.
+        """
+
+        # during initialization, load weights from the ema model
+        if (self.step_counter == self.start_iter
+                and self.generator_ema is not None):
+            if is_module_wrapper(self.generator):
+                self.generator.module.load_state_dict(
```

Comment:
Please consider this situation in the unit test

Reply:
The first `if` is considered. For the second one, it is True only when distributed/multiple GPU learning is used, and therefore I think it is not testable here.